src/dbnode/storage/series/series.go

```diff
@@ -288,6 +289,19 @@ func (s *dbSeries) Write(
 	annotation []byte,
 	wOpts WriteOptions,
 ) (bool, error) {
+	if wOpts.BootstrapWrite {
+		at := timestamp.Truncate(s.opts.RetentionOptions().BlockSize())
+		alreadyExists, err := s.blockRetriever.IsBlockRetrievable(at)
+		if err != nil {
+			return false, fmt.Errorf(
+				"error checking block retrievable for bootstrap write: %v", err)
+		}
+		if alreadyExists {
+			return false, fmt.Errorf(
+				"bootstrap write for block is retrievable: block_start=%s", at)
```

Comment:
meganit: this message doesn't read like an error, maybe reword it a little to `cannot write block at %s: already exists` or something along those lines?